### PR TITLE
Gutenberg: show opt-in modals on wpcalypso

### PR DESCRIPTION
**Pull request by gwwar**
#### Changes proposed in this Pull Request

* This enables the `gutenberg/opt-in` feature flag to enable Gutenberg opt-in modals to be shown on wpcalypso. We should see if we need to prepare any e2e test updates.

screen shot 2018-11-07 at 4 20 36 pm

screen shot 2018-11-07 at 4 20 43 pm

* This also changes the z-index used by the Gutenberg opt-in notice so it's placed behind the global notices (all viewports) and behind the sidebar (only mobile viewports)

#### Testing instructions

- Sandbox D20226-code, or add `enable-gutenberg` sticker to a simple site of your choice
- Checkout this branch, clicking on "Try the new editor" will take you to the Calypsoified Gutenberg Editor.
- Click on close takes you back to Calypso
- Folks can also change their preference with:
screen_shot_2018-11-07_at_4 01 49_pm

### CODE REVIEW

**Comment by gwwar** [comment]
screen shot 2018-11-07 at 4 59 53 pm

Saw these errors on the first full e2e run. Will try again to see if these are timeouts or if we need to prep an update.

**Comment by gwwar** [comment]
@Automattic/lannister anyone is welcome to land this after we have e2e tests prepared or if  folks let us know that this was a bad run.

**Comment by mmtr** [comment]
> @mmtr What if instead of changing the e2e tests, we changed the notice z-index to not appear in front of the sidebar? 🤔

Thanks @Copons! Nice work-around 😄. Applied in https://github.com/Automattic/wp-calypso/pull/28372/commits/b38a8659b242b24b58799c569032e1ea988ff051

The notice `z-index` matches now the one used by the sidebar when the sidebar is open in mobile devices, so it's placed behind it. The problem is not present in bigger viewports, as the notice is over the content side.

**Comment by mmtr** [comment]
E2E tests passing 🎉

@Automattic/lannister can we get a second review here checking my last changes for fixing the E2E tests?

File: client/post-editor/editor-gutenberg-opt-in-notice/style.scss [hunk 1]
```diff
@@ -41,6 +41,13 @@
 	}
 }
 
+@include breakpoint( '<660px' ) {
+	// We place the notice behind the sidebar on mobile devices
+	.focus-sidebar .editor-gutenberg-opt-in-notice {
+		z-index: z-index( 'root', '.editor-sidebar' );
```
Comment (Copons) [inline client/post-editor/editor-gutenberg-opt-in-notice/style.scss]: In hindsight, I'm not super happy about this. 🤔 
What do you think if, instead, we completely `display: none` the notice when the sidebar is open?

**Pull request merged**